### DISCUSSION TITLE
Allow tagging transactions in POS

### DISCUSSION THREAD

**Discussion by Kixunil** [btcpayserver/btcpayserver#2988] [Ideas / Feature Requests]
## Problem

When accepting payments through POS in physical stores sometimes a transaction is special. Examples are:

* discounts for various reasons - people using affiliate coupons, loyalty cards...
* test transactions - e.g. if a payment fails the merchant may want to check if the problem is with his node or customer wallet by paying an invoice to himself. Such invoice obviously should **not** go to accounting.

However it is currently required for merchants to either remember which transaction was which kind or track it using a separate tool (paper, spreadsheet...) This is inconvenient and error-prone. It also requires manual post-processing of transactions before they can go to accounting.

## Proposed solution

Allow tagging transactions in POS. When setting up the POS app, the merchant can pick which tags can be added from POS. Then GUI similar to GitHub labels is shown in POS allowing the merchant to select a tag (or multiple tags). Those are stored with transaction and exported in CSV export tool allowing the merchant to filter them etc.

**Comment by bolatovumar** [comment]
@Kixunil We already have a way to tag transactions and add comments to them in the wallet transaction list. Is this what you want or did you mean something different?

![Capture](https://user-images.githubusercontent.com/1934678/141703990-8b926244-5a43-4f3d-8203-d5e9e177a4cd.PNG)

**Reply by Kixunil** [reply]
I mean from POS directly. Doing it in the transaction list is not great because either the user has to go there after payment and change it (possibly giving bartenders etc admin permissions) or an admin has to add tags at the end of the day and remember which transaction is which (possibly writing it down somewhere extra). It can of course still use same infrastructure, just have some UI for it in POS and allow POS setting it when it creates transaction.

**Reply by bolatovumar** [reply]
I don't really understand where in the POS you want to add tags and what would the tags be added to. Can you show a screenshot of a place where you would want to add a tag?
